datafusion/physical-optimizer/src/limit_pushdown.rs

```diff
@@ -146,6 +146,14 @@ pub fn pushdown_limit_helper(
         global_state.skip = skip;
         global_state.fetch = fetch;
 
+        if limit_exec.input().as_any().is::<CoalescePartitionsExec>() {
+            // If the child is a `CoalescePartitionsExec`, we should not remove the limit
+            // the push_down through the `CoalescePartitionsExec` to each partition will not guarantee the limit.
+            // todo we may have a better solution if we can support with_fetch for limit inside CoalescePartitionsExec.
```

Comment:
TODO:

Reply:
Thank you @xudong963  for review, change the comments, and added a follow-up issue:
https://github.com/apache/datafusion/issues/14446